public/application.js

```diff
@@ -27,6 +28,14 @@ document.addEventListener('DOMContentLoaded', (event) => {
       isDisabled: function() {
         return this.isSending || !this.isValid();
       },
+      filteredSparkles: function() {
+        if(this.username === ""){
+          return this.recentSparkles
+        }
+        return this.recentSparkles.filter( s => s.sparklee === this.username)
```

Comment:
I love the usage of the [filter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter) function here. This is a great way to filter the data in the browser and implement a search feature.